wolfcrypt/src/asn.c

```diff
@@ -2467,10 +2485,12 @@ int GetObjectId(const byte* input, word32* inOutIdx, word32* oid,
     #ifdef ASN_DUMP_OID
         word32 i;
     #endif
-
         if (oidType != oidIgnoreType) {
             checkOid = OidFromId(*oid, oidType, &checkOidSz);
-
+            #ifdef WOLFSSL_OID_IPSEC_IKE
+            checkOid = OidFromId_conflict(checkOid, actualOid);
```

Comment:
Under this solution, it appears that we end up doing a XMEMCMP twice doing essentially the same check.  One XMEMCMP is inside OidFromId_conflict() comparing actualOid to checkOid.  The second is at the bottom of GetObjectId(), again comparing actualOid to checkOid.  There may be a more efficient solution here?

Reply:
GetObjetId also has the type parameter. Maybe we can subdivide the CertExt type to have a new type for IKE? That was the intention of the type field.

Reply:
Both are the same type 10 in this case.

Reply:
Put simpler way, though not cool. Give me your thoughts.